models/DataObject/ClassDefinition/Data/Wysiwyg.php

```diff
@@ -99,6 +99,10 @@ public function setMaxCharacters(int|string $maxCharacters): void
      */
     public function getDataForResource(mixed $data, DataObject\Concrete $object = null, array $params = []): ?string
     {
+        if (is_string($data) && !($params['no_sanitize'] ?? false)) {
```

Comment:
With the current approach the API works as follows:
```php
//documents (data is sanitized in the API without save)
$document = Document\Page::getById(225);
$editable = new Wysiwyg();
$editable->setName('mycontent');
$editable->setDataFromResource('delay of &lt; 2 minutes. <script>alert("Hey!..")</script>');
$document->setEditable($editable);
//$document->save();
p_r($document->getEditable('mycontent')->getValue()); //sanitized data

//dataobjects (data is NOT sanitized in the API without save)
$test = Test::getById(1167);
$test->setDescription('delay of &lt; 23 minutes.<script>alert("Hey!..")</script>');
//$test->save();
p_r($test->getDescription()); //unsanitized data

//translations (data is NOT sanitized in the API without save)
$translation = new Translation();
$translation->setKey('from API');
$translation->addTranslation('en', 'delay of &lt; 2 minutes. <script>alert("Hey!..")</script>');
//$translation->save();
p_r($translation->getTranslation('en')); //unsanitized data
```

@brusch Would that be feasible to keep the current behavior like this?

Reply:
I'd prefer to have the behavior consistent, also for documents. 
It should be possible to override `save()` method in `Wysiwyg` class and do the sanitation there, right? 

```php 
public function save()
{
    $this->text = $helper->sanitizeFor('body', $data);
    $this->getDao()->save(); 
}
```

Reply:
yes, this will work. Changes adapted.